File: Data Structures & Algorithms/eating-bananas/submission-3.cs
public class Solution {
    public int MinEatingSpeed(int[] piles, int h) {
        int low = 1, high = piles.Max();
        
        while (low <= high){
            int mid = low + (high-low) / 2;
            int time = 0;
            foreach (int pile in piles){
                time += (int)Math.Ceiling((double)pile / mid);
            }

            if (time <= h) high = mid-1;
            else low = mid+1;
        }

        return low;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int res = nums[0];
        int low = 0, high = nums.size()-1;

        while (low <= high){
            int mid = low + (high-low) / 2;
            res = min(res, nums[mid]);

            if (nums[mid] < nums[high]) high = mid-1;
            else low = mid+1;
        }

        return res;
    }
};
